src/WebJobs.Script.WebHost/Security/KeyManagement/DefaultSecretManagerProvider.cs

```diff
@@ -61,7 +61,7 @@ public sealed class DefaultSecretManagerProvider : ISecretManagerProvider
         internal ISecretsRepository CreateSecretsRepository()
         {
             string secretStorageType = Environment.GetEnvironmentVariable(EnvironmentSettingNames.AzureWebJobsSecretStorageType);
-            var storageConnection = _configuration.GetWebJobsConnectionStringSection(ConnectionStringNames.Storage);
+            var storageConnection = _azureStorageProvider.Configuration.GetWebJobsConnectionStringSection(ConnectionStringNames.Storage);
```

Comment:
Ideally, the storage provider wouldn't just expose the configuration and expect the logic to consume/validate is implemented by the consumer. We can eliminate the need to get configuration and expose the property by adding a method to check whether the provider is enabled (which, internally, would mean validating whether the connection string is setup) and replacing the `Exists` call below with that method instead.

Reply:
Added ConnectionExists public method to IAzureStorageProvider that attempts to create the BlobServiceClient and returns true/false.